src/Stratis.Features.FederatedPeg/PremineCoinbaseSplitter.cs

```diff
@@ -0,0 +1,28 @@
+﻿using NBitcoin;
+using Stratis.Features.FederatedPeg.Interfaces;
+
+namespace Stratis.Features.FederatedPeg
+{
+    public class PremineCoinbaseSplitter : ICoinbaseSplitter
+    {
+        /// <summary>
+        /// The number of outputs we break the premine reward up into, so that the federation can build more than one transaction at once.
+        /// </summary>
+        public const int FederationWalletOutputs = 10;
+
+        public void SplitReward(Transaction coinbase)
+        {
+            TxOut premineOutput = coinbase.Outputs[0];
+
+            Money newTxOutValues = premineOutput.Value / FederationWalletOutputs;
```

Comment:
This will burn all premine funds if `premineOutput.Value < FederationWalletOutputs`. And we will lose funds if `premineOutput.Value > FederationWalletOutputs && (premineOutput.Value % FederationWalletOutputs > 0)`.